tm2/pkg/bft/consensus/wal_generator.go

```diff
@@ -36,7 +36,9 @@ func WALGenerateNBlocks(t *testing.T, wr io.Writer, numBlocks int) (err error) {
 	config := getConfig(t)
 
 	app := kvstore.NewPersistentKVStoreApplication(filepath.Join(config.DBDir(), "wal_generator"))
-	defer app.Close()
+	defer func() {
+		err = app.Close()
+	}()
```

Comment:
This will overwrite any errors of the function on `return`; please revert to before (same for other instances on this file)

Reply:
done